scripts/new-version-pr-generator.js

```diff
@@ -0,0 +1,154 @@
+
+const releaseVersion = `v${PACKAGE_VERSION}`;
+
+function isMajorPrerelease(version) {
+  return (
+    semver.prerelease(version) &&
+    semver.minor(version) === 0 &&
+    semver.patch(version) === 0
+  );
+}
+
+const baseBranch = isMajorPrerelease(PACKAGE_VERSION)
+  ? `v${semver.major(PACKAGE_VERSION)}`
+  : 'master';
+
+const polarisBotName = 'Shopify Polaris Bot';
+const polarisBotEmail = 'shopify-polaris-bot@users.noreply.github.com';
+const polarisBotToken = secrets.github['shopify-polaris'];
+const shopifyPolarisBotGitOverride = `GIT_COMMITTER_NAME='${polarisBotName}' GIT_COMMITTER_EMAIL='${polarisBotEmail}'`;
+
+function pullRequestTemplate(repository, version) {
+  return `
+## Version ${version} of @shopify/polaris just got published!
+
+See what’s new: https://github.com/Shopify/polaris-react/releases/tag/${version}
+
+cc @kaelig @dfmcphee @amrocha @BPScott
+
+---
+
+<details>
+<summary>🚨 What to do if you see “Your tests failed on CircleCI”?</summary>
+
+If tests fail, you may have to troubleshoot the problem locally.
+
+1. Checkout the \`update-polaris-${version}\` branch:
+    \`\`\`bash
+    dev cd ${repository}
+    git fetch
+    git checkout update-polaris-${version}
+    dev up
+    \`\`\`
+1. Apply changes/fixes
+1. Commit
+1. Push your work to the \`update-polaris-${version}\` branch:
+    \`\`\`bash
+    git push origin update-polaris-${version}
+    \`\`\`
+1. Repeat until the build & tests go ✅
+1. 🎩
+1. :shipit:
+</details>
+`.trim();
+}
+
+function failedUpdateMessage(repository, version) {
+  return `It seems your automatic branch creation for ${repository} failed. This can be due to many reasons. To resolve this follow these steps:
+
+  1. Checkout "${repository}"
+  2. Get the latest version of master "git checkout master && git pull"
+  3. Create a new branch based on the version that failed "git checkout -b update-polaris-v${version}"
+  4. Update "shopify/polaris" by running "npx yarn upgrade @shopify/polaris@${version}"
+  5. Add the changed files to git by running "git add package.json yarn.lock"
+  6. Commit the changes with "git commit -m 'Update @shopify/polaris to ${version}'
+  7. Push the changes with "git push origin update-polaris-v${version}"
+  8. Create a new pull request for the branch in "${repository}"`;
+}
+
+mkdir(sandbox);
+
+const tasks = repositories.map((repository) => {
+  return new Promise((resolve, reject) => {
+    try {
+      const repositoryDirectory = pathResolve(sandbox, repository);
+
+      // Clone the repositories to the sandbox directory
+      Retry(
+        execSync(
+          `git clone --branch ${baseBranch} --single-branch https://${polarisBotToken}@github.com/Shopify/${repository}.git`,
+          {
+            cwd: sandbox,
+            stdio: 'inherit',
+          },
+        ),
+      );
+
+      const commands = [
+        `git checkout -b update-polaris-${releaseVersion}`,
+        'npm config set @shopify:registry https://registry.npmjs.org',
```

Comment:
I don't know if Shopify/web has dependencies that live on our private packagecloud, but if so, this line could make the upgrade fail.

Reply:
Can we just use the npm registry? Is there any reason we specify the @shopify one?

Reply:
By default, `@shopify`-scoped are querying Shopify's private packagecloud registry. 

I believe this line is changing this setting because "yarn upgrade" would fail due to certain packages being both on the private and the public registries but not all versions would be available on the private one. 

It may not be an issue anymore, so this is worth testing to make sure.

Reply:
This was already here before, Alex just moved it from the old script, it works

Reply:
Yeah lets remove it and give it a go 👍

Reply:
>This was already here before, Alex just moved it from the old script, it works

Yes it was already there, and it works when running `yarn upgrade` on `Shopify/polaris-styleguide`, but there's a good chance it'll make `yarn upgrade` fail on `Shopify/web` because it also depends on private packages.